UnityProject/Assets/Scripts/Chat/DiscordWebhookMessage.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System.Collections.Specialized;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Net;
+using System.Text.RegularExpressions;
+using DatabaseAPI;
+
+
+namespace DiscordWebhook
+{
+	public class DiscordWebhookMessage : MonoBehaviour
+	{
+		private static DiscordWebhookMessage instance;
+		public static DiscordWebhookMessage Instance => instance;
+
+		private Queue<string> OOCMessageQueue = new Queue<string>();
+		private Queue<string> AdminAhelpMessageQueue = new Queue<string>();
+		private Queue<string> AnnouncementMessageQueue = new Queue<string>();
+		private Queue<string> AllChatMessageQueue = new Queue<string>();
+		private float SendingTimer = 0;
+		private const float MessageTimeDelay = 1f;
+
+		private void Awake()
+		{
+			if (instance == null)
+			{
+				instance = this;
+			}
+			else
+			{
+				Destroy(this);
+			}
+		}
+
+		private void Update()
+		{
+			SendingTimer += Time.deltaTime;
+			if (SendingTimer > MessageTimeDelay)
+			{
+				var URL = GetUrl(DiscordWebhookURLs.DiscordWebhookOOCURL);
+				FormatMessage(URL.Item1, URL.Item2);
+
+				URL = GetUrl(DiscordWebhookURLs.DiscordWebhookAdminURL);
+				FormatMessage(URL.Item1, URL.Item2);
+
+				URL = GetUrl(DiscordWebhookURLs.DiscordWebhookAnnouncementURL);
+				FormatMessage(URL.Item1, URL.Item2);
+
+				URL = GetUrl(DiscordWebhookURLs.DiscordWebhookAllChatURL);
+				FormatMessage(URL.Item1, URL.Item2);
```

Comment:
This is entirely repeated code. At this point you probably want a dictionary of <your enum, queue> or something to loop through.

Reply:
Made dict